backend/conspects/serializers.py

```diff
@@ -67,15 +80,36 @@ def validate(self, data):
 
     def get_files(self, obj):
         files = obj.folder_files.all()
-        return FileSerializer(files, many=True).data
+        return FileSerializer(files, many=True, context=self.context).data
+
+    def get_user_permission(self, obj):
+        user = self.context['request'].user
+        if user.is_authenticated:
+            if UserEdition.objects.filter(user=user, edition__isnull=True, permission_type='admin').exists():
+                return "admin"
+            permission = UserEdition.objects.filter(user=user, edition=obj.edition).first()
+            return permission.permission_type if permission else None
+        return None
 
 
 class FileSerializer(serializers.ModelSerializer):
+    user_permission = serializers.SerializerMethodField(read_only=True)
+
     class Meta:
         model = File
-        fields = ["id", "name", "extension", "content", "can_be_edited", "can_be_previewed", "is_attachment"]
+        fields = ["id", "name", "extension", "content", "can_be_edited", "can_be_previewed", "is_attachment",
+                  "user_permission"]
         read_only_fields = ["can_be_edited", "can_be_previewed", "is_attachment"]
 
+    def get_user_permission(self, obj):
+        user = self.context['request'].user
+        if user.is_authenticated:
+            if UserEdition.objects.filter(user=user, edition__isnull=True, permission_type='admin').exists():
+                return "admin"
+            permission = UserEdition.objects.filter(user=user, edition=obj.folder.edition).first()
+            return permission.permission_type if permission else None
+        return None
```

Comment:
There are three similar definitions of this method? couldn't it be extracted into another helper function and ```get_user_permission``` be a wrapper?

Reply:
Fair point, it could be extracted